src/components/TldrawCanvas.tsx

```diff
@@ -1047,15 +1047,27 @@ const TldrawCanvas = ({ title }: Props) => {
                 return TLSelectTool.children().map((c) => {
                   if (c.id === "dragging_handle") {
                     const Handle = c as unknown as typeof DraggingHandle;
+                    const allRelationIdSet = new Set(allRelationIds);
+                    const allAddReferencedNodeActionsSet = new Set(
+                      allAddReferencedNodeActions
+                    );
                     return class extends Handle {
                       override onPointerUp: TLPointerEvent = async () => {
                         this.onComplete({
                           type: "misc",
                           name: "complete",
                         });
 
-                        const arrow = this.app.getShapeById(this.shapeId);
-                        if (!arrow) return;
+                        const shape = this.app.getShapeById(this.shapeId);
+                        if (!shape) return;
+                        // TODO: find a better way to identify custom arrow shapes
+                        // shape.type or shape.name probably?
```

Comment:
could use `roamAlphaAPI.pull` but we already have the set in scope so might as well use it as you have it